src/util/beacon.ts

```diff
@@ -6,6 +6,8 @@ import { SendBeaconResult } from "../@types"
  */
 const hasBeaconSupport = "sendBeacon" in navigator
 
+export type BeaconMethod = "GET" | "POST"
```

Comment:
Not a blocker, but i'd consider using a [string enum](https://www.typescriptlang.org/docs/handbook/enums.html#string-enums) here instead, as we know the values are finite.
```suggestion
export enum BeaconMethod {
 Get = "GET",
 Post = "POST",
} 
```